### PR TITLE
clatd: init at 1.6

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

A CLAT component for 464XLAT: https://github.com/toreanderson/clatd

I will do NixOS module later on and proper testing.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/clatd/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "toreanderson";
+    repo = "clatd";
+    rev = "v${version}";
+    hash = "sha256-ZUGWQTXXgATy539NQxkZSvQA7HIWkIPsw1NJrz0xKEg=";
+  };
+
+  makeFlags = [
+    "DESTDIR="
```
Comment (figsoda) [inline pkgs/tools/networking/clatd/default.nix:18]: this is the default
```suggestion
```

File: pkgs/tools/networking/clatd/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "clatd";
+  version = "1.6";
+
+  src = fetchFromGitHub {
+    owner = "toreanderson";
+    repo = "clatd";
+    rev = "v${version}";
+    hash = "sha256-ZUGWQTXXgATy539NQxkZSvQA7HIWkIPsw1NJrz0xKEg=";
+  };
+
```
Comment (figsoda) [inline pkgs/tools/networking/clatd/default.nix:16]: the shebang needs to be patched, this does the trick for me, though idk if this is the idiomatic way to package perl stuff

```suggestion

  buildInputs = [
    (perl.withPackages (p: [ p.NetDNS p.NetIP ]))
  ];

```

File: pkgs/tools/networking/clatd/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
```
Comment (figsoda) [inline pkgs/tools/networking/clatd/default.nix:4]:
```suggestion
, perl
}:
```

**Review by figsoda — CHANGES_REQUESTED** [review]

**Comment by hax404** [comment]
I built a test for clatd some time ago. Feel free to use. https://github.com/hax404/nixpkgs/commit/6be04700ef7f5441b44102add9dca95b681f7c45

**Comment by RaitoBezarius** [comment]
Please ignore this PR given the existence of superior https://github.com/NixOS/nixpkgs/pull/240982 containing a clatd component AFAIK.

File: pkgs/tools/networking/clatd/default.nix (hunk 2 above)
Comment (RaitoBezarius) [reply pkgs/tools/networking/clatd/default.nix:16]: Grepping shows that it is the case I believe.

**Comment by RaitoBezarius** [comment]
Reopened because Jool is actually inferior in practical usecases.

**Pull request closed without merging**